src/regex.go

```diff
@@ -19,7 +33,13 @@ func findNamedRegexMatch(pattern, text string) map[string]string {
 }
 
 func findAllNamedRegexMatch(pattern, text string) []map[string]string {
-	re := regexp.MustCompile(pattern)
+	re := regexCache[pattern]
+	if re == nil {
```

Comment:
this is the same functionality twice, can we split this in a separate function?

Reply:
Yes, I put the compile and cache in a separate functions